docs/isochrone/java.md

```diff
@@ -0,0 +1,134 @@
+queryGraph.lookup(Collections.singletonList(qr));
+
+HintsMap hintsMap = new HintsMap();
+initHints(hintsMap, hReq.getParameterMap());
+Weighting weighting = hopper.createWeighting(hintsMap, encoder, graph);
+Isochrone isochrone = new Isochrone(queryGraph, weighting, reverseFlow);
+if (distanceInMeter > 0) {
+    double maxMeter = 50 * 1000;
+    if (distanceInMeter > maxMeter)
+        throw new IllegalArgumentException("Specify a limit of less than " + maxMeter / 1000f + "km");
+    if (buckets > (distanceInMeter / 500))
+        throw new IllegalArgumentException("Specify buckets less than the number of explored kilometers");
+    isochrone.setDistanceLimit(distanceInMeter);
+} else {
+    long maxSeconds = 80 * 60;
+    if (timeLimitInSeconds > maxSeconds)
+        throw new IllegalArgumentException("Specify a limit of less than " + maxSeconds + " seconds");
+    if (buckets > (timeLimitInSeconds / 60))
+        throw new IllegalArgumentException("Specify buckets less than the number of explored minutes");
+
+    isochrone.setTimeLimit(timeLimitInSeconds);
+}
+
+List<List<Double[]>> list = isochrone.searchGPS(qr.getClosestNode(), buckets);
+if (isochrone.getVisitedNodes() > hopper.getMaxVisitedNodes() / 5) {
+    throw new IllegalArgumentException("Reset: too many junction nodes would have to explored (" + isochrone.getVisitedNodes() + "). You may need to increase this value.");
+}
+
+int counter = 0;
+for (List<Double[]> tmp : list) {
+    if (tmp.size() < 2) {
+        throw new IllegalArgumentException("Too few points found for bucket " + counter + ". "
+                + "Please try a different point, a smaller buckets count, or a larger time limit.. ");
+    }
+    counter++;
+}
+```
+
+Finally, convert the isochrone into a result, for easy return. You can either return a point list or a polygon.
+```java
+Object calcRes;
+
+// get result as a point list
+if ("pointlist".equalsIgnoreCase(resultStr)) {
+    calcRes = list;
+
+// or get result as a polygon
+} else if ("polygon".equalsIgnoreCase(resultStr)) {
+    list = rasterHullBuilder.calcList(list, list.size() - 1, rasterDistance, bufferDistance, quadrantSegments);
+
+    ArrayList polyList = new ArrayList();
+    int index = 0;
+    for (List<Double[]> polygon : list) {
+        HashMap<String, Object> geoJsonMap = new HashMap<>();
+        HashMap<String, Object> propMap = new HashMap<>();
+        HashMap<String, Object> geometryMap = new HashMap<>();
+        polyList.add(geoJsonMap);
+        geoJsonMap.put("type", "Feature");
+        geoJsonMap.put("properties", propMap);
+        geoJsonMap.put("geometry", geometryMap);
+
+        propMap.put("bucket", index);
+        geometryMap.put("type", "Polygon");
+        // we have no holes => embed in yet another list
+        geometryMap.put("coordinates", Collections.singletonList(polygon));
+        index++;
+    }
+    calcRes = polyList;
+} else {
+    throw new IllegalArgumentException("type not supported:" + resultStr);
+}
```

Comment:
I'm not sure if we should duplicate the whole servlet code. Why not show really only a simple time limit example that returns the point lists and link to the full code in the servlet?